lib/Core/Search/Common/EventSubscriber/ContentEventSubscriber.php

```diff
@@ -55,26 +59,24 @@ public function onCopyContent(CopyContentEvent $event): void
 
     public function onBeforeDeleteContent(BeforeDeleteContentEvent $event): void
     {
+        $contentLocations = $this->locationService->loadLocations($event->getContentInfo());
         try {
-            $event->getContentInfo()->getMainLocation()?->parentLocationId;
+            foreach ($contentLocations as $contentLocation){
+                $this->contentParentLocations[] = $contentLocation->parentLocationId;
```

Comment:
Even as PHP is single-threaded and and `onBeforeDeleteContent` should not be called again before it's `onDeleteContent` is called, let's not assume that. We should use Content ID as key here, and then unset it in `onDeleteContent` after the message is dispatched.

Also, to be extra specific the property can be called `parentLocationIdsByContentId`.

Reply:
resolved in https://github.com/netgen/ibexa-search-extra/pull/9/commits/c7788685c4049ca5a31a9ae450be1ee56b04cef8

Reply:
Good, but please use `unset($this->parentLocationIdsByContentId[$event->getContentInfo()->id])` at the end.

Reply:
fixed in 7e34ad4302ec53504ae345759b7d4ee6b3e8ae85